### PR TITLE
feat: add flag to support Enterprise GitHub 

### PR DESCRIPTION
**Pull request by ahothan**
The --github-url flag can be used to operate on repositories hosted on an Enterprise Github server.
See the doc update for more information.
---
Tested on my work GitHub Enterprise Server 3.5.7 and working fine.

### CODE REVIEW

**Comment by ahothan** [comment]
addresses https://github.com/dailymotion-oss/octopilot/issues/164

File: main.go [hunk 1]
```diff
@@ -49,6 +49,7 @@ func init() {
 	pflag.Int64Var(&options.GitHub.InstallationID, "github-installation-id", int64(getenvInt("GITHUB_INSTALLATION_ID")), "For the `app` GitHub auth method, contains the GitHubApp Installation ID. Default to the GITHUB_INSTALLATION_ID env var.")
 	pflag.StringVar(&options.GitHub.PrivateKey, "github-privatekey", os.Getenv("GITHUB_PRIVATEKEY"), "For the `app` GitHub auth method, contains the GitHubApp Private key file in PEM format. Default to the GITHUB_PRIVATEKEY env var.")
 	pflag.StringVar(&options.GitHub.PrivateKeyPath, "github-privatekey-path", os.Getenv("GITHUB_PRIVATEKEY_PATH"), "For the `app` GitHub auth method, contains the GitHubApp Private key file path `/some/key.pem` (used if the github-privatekey is empty). Default to the GITHUB_PRIVATEKEY_PATH env var.")
+	pflag.StringVar(&options.GitHub.Url, "github-url", "https://github.com", `GitHub server URL`)
```
Comment (vbehar) [inline main.go:52]:
```suggestion
	pflag.StringVar(&options.GitHub.Url, "github-url", repository.PublicGithubUrl, `GitHub server URL`)
```

File: repository/git.go [hunk 2]
```diff
@@ -16,15 +17,19 @@ import (
 )
 
 func cloneGitRepository(ctx context.Context, repo Repository, localPath string, options GitHubOptions) (*git.Repository, error) {
-	url := fmt.Sprintf("https://github.com/%s.git", repo.FullName())
+	gitUrl, err := url.JoinPath(options.Url, repo.GitFullName())
```
Comment (vbehar) [inline repository/git.go:20]: url.JoinPath has been added in go 1.19, so to use it you'll need to update the version of go in go.mod (it's used automatically in the github actions to install the same version of go for the CI jobs)

**Review by vbehar — COMMENTED** [review]
thanks, good work! if you can just fix the pipelines, then I'll merge it

File: main.go (hunk 1 above)
Comment (ahothan) [reply main.go]: good catch

File: repository/git.go (hunk 2 above)
Comment (ahothan) [reply repository/git.go:20]: done

File: .github/workflows/static-analysis.yml [hunk 3]
```diff
@@ -13,4 +13,4 @@ jobs:
       - name: Run linters
         uses: golangci/golangci-lint-action@v3.4.0
         with:
-          version: v1.47.2
+          version: v1.51.2
```
Comment (vbehar) [inline .github/workflows/static-analysis.yml:16]: you should also fix the .golangci.yml file to replace 1.18 with 1.19, and maybe fix a few warnings. I can have a look tomorrow if you prefer

Comment (ahothan) [reply .github/workflows/static-analysis.yml:16]: done

**Comment by vbehar** [comment]
ok, seems like the linters are still using an older go version. from https://github.com/golangci/golangci-lint-action I see that 

> v3.0.0+ requires explicit setup-go installation step prior to using this action: uses: actions/setup-go@v3.

and in https://github.com/dailymotion-oss/octopilot/blob/master/.github/workflows/static-analysis.yml we don't use setup-go...  you can copy these few lines from https://github.com/dailymotion-oss/octopilot/blob/master/.github/workflows/pr-build.yml for example (with `go-version-file: 'go.mod'`) and paste them in the static-analysis.yml file before the golangci action.

and then it should be ok!

thanks again for your work, and sorry that you had to go through all that...

**Comment by ahothan** [comment]
the linter still did not pick go 1.19, any suggestion to fix this?

**Comment by ahothan** [comment]
@vbehar just checking if we can merge this soon? I will need this soon as I'm currently using my forked binary

**Comment by vbehar** [comment]
> @vbehar just checking if we can merge this soon? I will need this soon as I'm currently using my forked binary

yes sorry I was blocked by a broken integration with codecov, it's fixed now, so I'm merging it right now

**Review by vbehar — DISMISSED** [review]

**Comment by vbehar** [comment]
argh sorry 1 more thing that blocks the merge:
> The base branch requires all commits to be signed. [Learn more about signing commits.](https://docs.github.com/articles/about-gpg/)

can you please sign your commit? thanks

**Comment by ahothan** [comment]
@vbehar I think it is signed now as I see the Verified marker on the merge

**Review by vbehar — APPROVED** [review]

**Comment by vbehar** [comment]
> The base branch requires all commits to be signed.

arghh it requires all commits to be signed, sorry you'll have to squash your commits together to sign the new commit. or if you want I can do it for you

**Comment by ahothan** [comment]
@vbehar will be best if you can do it, otherwise let me know

**Review by vbehar — APPROVED** [review]

**Review by vbehar — APPROVED** [review]

**Pull request merged**